src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointMessageWriter.cs

```diff
@@ -52,6 +54,12 @@ public MultiEndpointMessageWriter(IReadOnlyCollection<ServiceEndpoint> targetEnd
 
     public Task WriteAsync(ServiceMessage serviceMessage)
     {
+        if (serviceMessage is ClientInvocationMessage invocationMessage)
+        {
+            // Accroding to target endpoints in method `WriteMultiEndpointMessageAsync`
+            _clientInvocationManager.Caller.SetAckNumber(invocationMessage.InvocationId, TargetEndpoints.Length);
```

Comment:
when `TargetEndpoints.Length` is 0, the result is OK?

Reply:
class `AckHandler` could handle such condition correctly. See the logic of its method `SetExpectedCount`

Reply:
fix here https://github.com/Azure/azure-signalr/pull/2154/files#diff-1f07592019f7c2af9fcce1e14fae72fe2c02010db65cb61d9c822e4dda56d44cR62-R70